client/src/hooks/graphql/use-with-setup.tsx

```diff
@@ -189,23 +189,31 @@ export function useWithSetup(
   }, [trainError]);
 
   function nextStep() {
-    if (!status || idx === steps.length - 1) {
+    if (!status) {
       return;
     }
     if (isMentorEdited) {
       saveMentorDetails();
     }
-    setIdx(idx + 1);
+    if (idx === steps.length - 1) {
```

Comment:
```typescript
function addToIdx(delta: int = 1): void {
  // we have to add steps.length below because stupid js
  // returns negative mods, e.g.
  //    (0 - 1) % 10 == -1 // should be 9
  setIdx((!isNaN(Number(idx)? Number(idx) + (delta + steps.length): 0) % steps.length);
}
```